hack/generator/pkg/config/type_transformer.go

```diff
@@ -17,65 +18,110 @@ import (
 type TransformTarget struct {
 	PackagePath string `yaml:",omitempty"`
 	Name        string `yaml:",omitempty"`
+	Map         *MapType
+}
+
+type MapType struct {
+	Key   TransformTarget `yaml:",omitempty"`
+	Value TransformTarget `yaml:",omitempty"`
 }
 
 // A TypeTransformer is used to remap types
 type TypeTransformer struct {
 	TypeMatcher `yaml:",inline"`
 
+	// Property is a wildcard matching specific properties on the types selected by this filter
+	Property      string `yaml:",omitempty"`
+	propertyRegex *regexp.Regexp
+
 	Target TransformTarget
 
 	// This is used purely for "caching" the actual astmodel type after Initialize()
 	targetType astmodel.Type
 }
 
+func produceTargetType(target TransformTarget, descriptor string) (astmodel.Type, error) {
+	var result astmodel.Type
+
+	if target.Name != "" {
+		if target.PackagePath == "" {
+			primitive, err := primitiveTypeTarget(target.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			result = primitive
+		} else {
+			result = astmodel.MakeTypeName(
+				astmodel.MakePackageReference(target.PackagePath),
+				target.Name)
+		}
+	}
+
+	if target.Map != nil {
+		if result != nil {
```

Comment:
minor: It might be clearer if rather than the result + check here, you just did:
```
if target.Map != nil && target.Name !=nil {
     return errors.Errorf()
}
```

up at the top?

It saves you a `result` (you can just return inline like you were), and it moves one of the base failure cases to the top?

Reply:
yeah this is so much clearer, ugh 😛